libkbfs/folder_block_ops.go

```diff
@@ -1001,6 +1027,37 @@ func (fbo *folderBlockOps) ClearCachedDirEntry(lState *lockState, dir path) {
 	// in there as well.)
 }
 
+func (fbo *folderBlockOps) updateDirtyEntryLocked(
+	ctx context.Context, lState *lockState, de DirEntry) (
+	updated bool, newDe DirEntry) {
+	fbo.blockLock.AssertAnyLocked(lState)
+
+	refDe, ok := fbo.deCache[de.Ref()]
+	if !ok {
+		return false, de
+	}
+
+	// Only update the entry if there's a full directory update,
+	// or a mtime/ctime update.
+	if !refDe.dirEntry.IsInitialized() && refDe.dirEntry.Mtime == 0 &&
+		refDe.dirEntry.Ctime == 0 {
+		return false, de
+	}
+
+	if refDe.dirEntry.IsInitialized() {
+		return true, refDe.dirEntry
+	}
+
+	// Just update the times.
+	if refDe.dirEntry.Mtime > 0 {
```

Comment:
In `addDirEntryInCacheLocked()`, we update the `Ctime` and `Mtime` of the parent directory, but we don't necessarily have the full `DirEntry` for it (which would need to be fetched from the grandparent block).